tests-legacy/PrestaShopBundle/Utils/DatabaseCreator.php

```diff
@@ -47,17 +47,27 @@ public static function createTestDB()
         define('_PS_MODULE_DIR_', _PS_ROOT_DIR_ . '/tests-legacy/resources/modules/');
         require_once __DIR__ . '/../../../install-dev/init.php';
 
-        $install = new Install();
+        $install = new class() extends Install {
+            public function setError($errors)
```

Comment:
Why you don't do it in Install class?

Reply:
This is in `tests-legacy` where there is no standard logger/error handler.

Reply:
instead of exit here, why not exit when you really need to. You break something I think.
if in the code it does:
```
$this->setError('blabla because of');
$this->setError('property is not defined');
$this->restoreDatabaseSettings();
```
It ends directly and exit execute the latest method.
You take the problem in the wrong way.

Reply:
`setError` accepts an array exactly for these purposes :)

I belive this is fine for `tests-legacy` and it already caught an error that was previously silenced. Please let me know how to fix the CI error.

Reply:
It's quite ugly, I'm sorry but I'm against doing it like that

Reply:
I fixed it this way, found a hidden issue with it, and propose this improvement. I belive, it is good and very well written.

If you think of a better way, please submit a PR against my branch or feel free to push to it directly.

Reply:
I'm sorry but if you create a pull request you expect to receive review otherwise don't create a pull request and use your own fork :sweat_smile: 
I'll not submit a PR on your branch or push it directly.

Reply:
Please tell me how you want this code to be fixed. Also, what are the PHP people behind PS and is there a good way to reach them on Slack, Discord or Skype?

Reply:
As I said [above](https://github.com/PrestaShop/PrestaShop/pull/23891#discussion_r609714584), you have to be careful when using exit. Using anonymous class is not the solution.

Reply:
This is how it was done - https://github.com/PrestaShop/PrestaShop/pull/23891/files#diff-30135fc378f8bb959681930c77e7f2135a09b6c9af8916429ed8ba3198bb4472L56

```
$this->setError('blabla because of');
$this->setError('property is not defined');
$this->restoreDatabaseSettings();
```

Is not a usecase we need for CI, `setError` is ok to fail on the first error in this legacy utility. My point is that previously all errors, even the legit ones, were silently ignored. Is it ok to fix this now by meging this in and allowing other people improve later if someone wants?